ion-schema/src/system.rs

```diff
@@ -862,27 +929,65 @@ impl Resolver {
+                    match isl_version {
+                        IonSchemaLanguageVersion::V1_0 => {
+                            // verify if there are any constraints with ISL 2.0 for this isl_type
+                            let has_isl_2_0_constraints =
+                                isl_type.constraints().iter().any(|c| match c.constraint {
+                                    IslConstraintImpl::Regex(_)
+                                    | IslConstraintImpl::Element(_)
+                                    | IslConstraintImpl::Annotations(_)
+                                    | IslConstraintImpl::Scale(_) => {
+                                        c.version == IonSchemaLanguageVersion::V2_0
+                                    }
+                                    _ => false,
+                                });
+
+                            if has_isl_2_0_constraints {
+                                return invalid_schema_error(format!("ISL type: {} contains constraints from ISL 2.0. Only use ISL 1.0 constraints for this method.", named_isl_type.name().as_ref().unwrap()));
+                            }
+                        }
+                        IonSchemaLanguageVersion::V2_0 => {
+                            // verify if there are any constraints with ISL 1.0 for this isl_type
+                            let has_isl_1_0_constraints =
+                                isl_type.constraints().iter().any(|c| match c.constraint {
+                                    IslConstraintImpl::Regex(_)
+                                    | IslConstraintImpl::Element(_)
+                                    | IslConstraintImpl::Annotations(_)
+                                    | IslConstraintImpl::Scale(_) => {
+                                        c.version == IonSchemaLanguageVersion::V2_0
+                                    }
+                                    _ => false,
+                                });
+
+                            if has_isl_1_0_constraints {
+                                return invalid_schema_error(format!("ISL type: {} contains constraints from ISL 1.0. Only use ISL 2.0 constraints for this method.", named_isl_type.name().as_ref().unwrap()));
+                            }
+                        }
+                    }
+
                     // convert IslType to TypeDefinition
                     let type_id: TypeId =
                         TypeDefinitionImpl::parse_from_isl_type_and_update_pending_types(
+                            isl_version,
                             named_isl_type,
                             type_store,
                             pending_types,
                         )?;
                 }
-                IslType::Anonymous(_) => {
+                IslTypeKind::Anonymous(_) => {
                     unreachable!("Top level ISL type definitions are always named type definitions")
                 }
```

Comment:
If this is truly unreachable, then I don't think you need to match on `kind` at all.

Reply:
Sure,I will change this.